test/controllers/actions/SubmittedDataRetrievalActionImplSpec.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.actions
+
+import base.SpecBase.{businessId, convertScalaFuture, taxYear}
+import cats.implicits._
+import controllers.actions.AuthenticatedIdentifierAction.User
+import controllers.actions.SubmittedDataRetrievalActionImplSpec._
+import models.common.JourneyAnswersContext
+import models.database.UserAnswers
+import models.journeys.Journey
+import models.requests.OptionalDataRequest
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+import play.api.libs.json.{JsObject, Json}
+import play.api.mvc.AnyContentAsEmpty
+import play.api.test.FakeRequest
+import stubs.repositories.StubSessionRepository
+import stubs.services.SelfEmploymentServiceStub
+import uk.gov.hmrc.auth.core._
+
+import java.time.Instant
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class SubmittedDataRetrievalActionImplSpec extends AnyWordSpecLike with Matchers {
```

Comment:
Would it be worth having some ITs for this class at some point? Could check what is in mongo and use wiremock for BE.

Reply:
Not sure.

I don't think we need many ITs, they slow down refactoring. If the codebase is young and changing, they don't give much value. But once we have Income and Expenses entire journey we may start thinking if there is a chance to catch more bugs by adding more integration tests. At the end of the day we have Regression Tests: Journey UI which will catch a lot of bugs and protect us from breaking the code.

I would say once we have stable codebase + 100% unit test coverage + Regression Journey UI tests, then we can start thinking on integration tests. Or altenatively, if the logic between 2 components is tricky we may also add it earlier.